src/Engine/ProtoAssociative/CodeGen.cs

```diff
@@ -351,6 +352,19 @@ private List<GraphNode> StrongConnectComponent(GraphNode node, ref int index, Di
                     continue;
                 }
 
+                bool equalIdentList = ProtoCore.AssociativeEngine.Utils.AreLHSEqualIdentList(node, subNode);
```

Comment:
@junmendoza , could you add a task to move some functions in `Utils` namespace to other places (for example, probably move directly to `AssociativeGraph` class), it will make these functions easier to be found, and more importantly, they are all graph related functions so that they should be in a closed namespace